walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -209,16 +286,114 @@ public void justEnoughRoom() {
+    // This second person should be ignored.
+    //
+    // Events  : |--A--|     |----A----|
+    //                 |--C--|
+    // Day     : |---------------------|
+    // Options :       |-----|
+    final String PERSON_C = "Person C";
+
+    Collection<Event> events = Arrays.asList(
+            new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+                    Arrays.asList(PERSON_A)),
+            new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TimeRange.END_OF_DAY, true),
+                    Arrays.asList(PERSON_A)),
+            new Event("Event 3", TimeRange.fromStartDuration(TIME_0830AM, 15),
+                    Arrays.asList(PERSON_C)));
 
     MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_C);
 
     Collection<TimeRange> actual = query.query(events, request);
     Collection<TimeRange> expected =
-        Arrays.asList(TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES));
+            Arrays.asList(TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void onlyOptionalAttendees() {
+    // Have two optional attendees, each with several gaps in their schedule. These gaps should be identified and
+    // returned as options for the meeting time.
+    // Events:      |--A--|                 |----A----|
+    //                          |--B--|     |----B----|
+    // Day:    |-------------------------------------------------|
+    //Options: |-----|    |-----|     |-----|         |----------|
+
+    final String OPTIONAL_PERSON_A = "Person A";
+    final String OPTIONAL_PERSON_B = "person B";
+
+    final Collection<Event> events = Arrays.asList(
+            new Event("Event 1", TimeRange.fromStartEnd(TIME_0800AM, TIME_0830AM, false),
+                    Arrays.asList(OPTIONAL_PERSON_A)),
+            new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TIME_0930AM, false),
+                    Arrays.asList(OPTIONAL_PERSON_B)),
+            new Event("Event 2", TimeRange.fromStartEnd(TIME_1000AM, TIME_1100AM, false),
+                    Arrays.asList(OPTIONAL_PERSON_A, OPTIONAL_PERSON_B))
+    );
+
+    MeetingRequest request = new MeetingRequest(new ArrayList<>(), 30);
+    request.addOptionalAttendee(OPTIONAL_PERSON_A);
+    request.addOptionalAttendee(OPTIONAL_PERSON_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected = Arrays.asList(
+            TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartDuration(TIME_0830AM, 30),
+            TimeRange.fromStartDuration(TIME_0930AM, 30),
+            TimeRange.fromStartEnd(TIME_1100AM, TimeRange.END_OF_DAY, true)
+    );
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void onlyOptionalAttendeesAllUnavailalbe() {
```

Comment:
Is this a deliberate misspelling?

Reply:
https://github.com/thomasogara/step-repository/pull/52/commits/f6ee446daa3b0f8c0d2b6eedcc770a6816abef72